test/e2e/openshift/openshift_test.go

```diff
@@ -85,6 +85,62 @@ var _ = Describe("Azure Container Cluster using the OpenShift Orchestrator", fun
 		}
 	})
 
+	It("should whitelist pods on compute nodes", func() {
+		namespaces := []string{
+			"openshift-infra",
+			"openshift-node",
+			"openshift-metrics",
+			"kube-",
+			"default",
+		}
+
+		whitelist := []string{
+			"sync-",
+			"ovs-",
+			"sdn-",
+		}
+		list, err := pod.GetAll("")
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, pod := range list.Pods {
+			skip := false
+			// we have a whitelist of namespaces that we will skip
+			for _, pattern := range namespaces {
+				if strings.Contains(pod.Metadata.Namespace, pattern) {
+					skip = true
+					break
+				}
+			}
+			if skip {
+				skip = false
+				continue
+			}
+
+			found := false
+			if strings.Contains(pod.Spec.Nodename, "compute") {
+				for _, pattern := range whitelist {
+					if strings.Contains(pod.Metadata.Name, pattern) {
+						found = true
+						break
+					}
+				}
+				Expect(found).To(BeTrue())
+			}
+		}
+	})
+
+	// Verify all pods have initialized, ready, podscheduled are true
+	It("should have all pods in ready state", func() {
+		list, err := pod.GetAll("")
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, pod := range list.Pods {
+			for _, cond := range pod.Status.Conditions {
+				Expect(cond.Status).To(BeTrue())
```

Comment:
too broad - just assert that Ready is true - leave the others

Reply:
it might be easier to check .status.phase is Running or Completed than iterate the conditions

Reply:
There are 3 condition types:
1.  Initialized
2.  Ready
3.  PodScheduled

All _should_ be `true`.  Would you prefer if we only check for `Ready`?

Reply:
I checked on a cluster with an application in a CrashLoopBackOff  state:
```
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2018-06-05T02:41:43Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2018-06-12T18:34:06Z
    message: 'containers with unready status: [glass]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: 2018-06-05T02:41:43Z
    status: "True"
    type: PodScheduled
```

The `phase` was set to `Running`.
```
  phase: Running
```

I believe that phase wouldn't be sufficient.